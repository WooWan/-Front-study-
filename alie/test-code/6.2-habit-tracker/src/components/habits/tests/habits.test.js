import renderer from "react-test-renderer";
import React from "react";
import Habits from "../habits";
describe("Habits component", () => {
  const habits = [
    { name: "sterling", count: 1, id: 1 },
    { name: "erling holland", count: 3, id: 2 },
  ];

  let HabitsComponent;
  let onIncrement;
  let onDecrement;
  let onDelete;
  let onAdd;
  let onReset;

  beforeEach(() => {
    onIncrement = jest.fn();
    onDecrement = jest.fn();
    onDelete = jest.fn();
    onAdd = jest.fn();
    onReset = jest.fn();
    HabitsComponent = (
      <Habits
        habits={habits}
        onIncrement={onIncrement}
        onDecrement={onDecrement}
        onDelete={onDelete}
        onAdd={onAdd}
        onReset={onReset}
      />
    );
  });

  it("render", () => {
    const component = renderer.create(HabitsComponent);
    expect(component.toJSON).toMatchSnapshot();
  });
});
