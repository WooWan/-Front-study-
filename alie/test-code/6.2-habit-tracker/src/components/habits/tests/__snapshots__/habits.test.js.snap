// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habits component render 1`] = `[Function]`;
